dotnet-interactive.sln

```diff
@@ -74,7 +74,9 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "docs", "docs", "{4DD50BCA-1
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.DotNet.Interactive.Http", "src\Microsoft.DotNet.Interactive.Http\Microsoft.DotNet.Interactive.Http.csproj", "{DE942DC3-B41B-45C5-9E01-3C4F042D3916}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.DotNet.Interactive.AspNetCore", "src\Microsoft.DotNet.Interactive.AspNetCore\Microsoft.DotNet.Interactive.AspNetCore.csproj", "{5167FC61-2FBF-4E0D-877E-C2DD354FB955}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.DotNet.Interactive.AspNetCore", "src\Microsoft.DotNet.Interactive.AspNetCore\Microsoft.DotNet.Interactive.AspNetCore.csproj", "{5167FC61-2FBF-4E0D-877E-C2DD354FB955}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SampleLibrary", "src\SampleLibrary\SampleLibrary.csproj", "{E9EF70C0-AB48-4D7B-B978-517FE76BEFB0}"
```

Comment:
This library looks more like test support than a sample that would be of any interest to someone browsing the repo. I'd nest it under the test project.

Reply:
Not sure what you mean.  If you're asking about the shape in the Solution Explorer, it's already under test:
![image](https://user-images.githubusercontent.com/926281/107431627-9085c500-6adb-11eb-8e64-63302330b67b.png)

If you're asking about where it lives on disk, we don't have a separate `test/` directory, but I can certainly create one.

Reply:
Mainly, the name is misleading. It's not a sample. Sometimes I put code like this inside the project folder, since it's really only of interest for a small subset of specific tests. 

Another approach that scales better is to use our test [utility](https://github.com/dotnet/interactive/blob/88a8ecb8fdd57513bea6eed3753be4da44568904/src/Microsoft.DotNet.Interactive.Tests/Utility/DirectoryUtility.cs#L48) that lets you declare file contents right in the test, and build this only when the test runs. It's more cohesive (you don't have to open up some other files to understand the test), more refactorable, and lets us add test cases more easily over time.

Reply:
Added a test helper that wraps that utility so sources can be specified on a per-test basis.